src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -203,17 +203,17 @@ class CompletePexEnvironment:
     def interpreter_search_paths(self) -> tuple[str, ...]:
         return self._pex_environment.interpreter_search_paths
 
-    def create_argv(
-        self, pex_filepath: str, *args: str, python: PythonExecutable | None = None
-    ) -> tuple[str, ...]:
+    def create_argv(self, pex_filepath: str, *args: str) -> tuple[str, ...]:
         pex_relpath = (
             os.path.relpath(pex_filepath, self._working_directory)
             if self._working_directory
             else pex_filepath
         )
         return (self._pex_environment.bootstrap_python.path, pex_relpath, *args)
 
-    def environment_dict(self, *, python: PythonExecutable | None = None) -> Mapping[str, str]:
+    def environment_dict(
+        self, *, python: PythonExecutable | PythonBuildStandaloneBinary | None = None
```

Comment:
This does get rid of the Python used to run the Pex PEX not being the `PEX_PYTHON_PATH`, which was the prior issue. But this seems to introduce a new issue or else highlight a pre-existing one. IIUC it is never safe / correct to run the Pex CLI (or any other Python program that builds Python dists) with PBS. @thejcannon Is it somehow constrained elsewhere that PBS will never be used by Pex to build dists? Maybe its the `--python-path` arg passed to the CLI that forces it (running under PBS), to only select other interpreters for locking and building PEXes since its guaranteed PBS is never a memeber of `--python-path`?

I think all these answers are likely affirmative, but reading the code it seems to me almost none of this is clear at all and comments might prove useful to avoid future issue refactoring this code.

Reply:
Yeah that's a good point. The Pex PEX is special so it should likely be treated as such. I'll stew on what that looks like